### PR TITLE
gocd: add deploy strategy

### PR DESCRIPTION
**Pull request by balboah**
#### What this PR does / why we need it:

Setting your deployment strategy for GoCD agents. One might not want the default 25% surge in new pods if they require a lot of resources to run.
 
#### Special notes for your reviewer:

The chart release number has been bumped twice because it will conflicts with #8521. If it's not a requirement for helm charts I would suggest deprecating such a changelog as its cumbersome to maintain. It's already tracked by git merges :)

#### Checklist
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

@arvindsv @GaneshSPatil @varshavaradarajan

### CODE REVIEW

File: stable/gocd/values.yaml [hunk 1]
```diff
@@ -168,6 +168,7 @@ server:
 agent:
   # agent.replicaCount is the GoCD Agent replicas Count. Specify the number of GoCD agents to run
   replicaCount: 0
+  deployStrategy: {}
```
Comment (varshavaradarajan) [inline stable/gocd/values.yaml:176]: Can you please add a comment above to explain and probably list out of deployment strategies' acceptable values?

Comment (balboah) [reply stable/gocd/values.yaml:176]: @varshavaradarajan done!

**Comment by arvindsv** [comment]
/ok-to-test

The changelog requirement seems to be from helm (and the bot) itself, I think.

**Pull request merged**